server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java

```diff
@@ -353,7 +355,10 @@ private MimeBodyPart createTextMsg(Mail originalMail) throws MessagingException
         }
         buffer.append(LINE_BREAK).append(LINE_BREAK);
         buffer.append("Error message:").append(LINE_BREAK);
-        buffer.append((String) originalMail.getAttribute("delivery-error")).append(LINE_BREAK);
+        AttributeUtils
+           .getValueAndCastFromMail(originalMail, AttributeName.of("delivery-error"), String.class)
```

Comment:
probably the String case should be handled by a specific method to reduce boilerplate